python/lsst/pipe/tasks/calibrateImage.py

```diff
@@ -817,6 +843,9 @@ def run(
                     "Cannot use do_illumination_correction with an image that has not had a flat applied",
                 )
 
+        if self.config.useButlerCamera:
+            self._update_camera_model(result.exposure, cameraModel)
+
```

Comment:
I might be nice to add a log message indicating the use of an updated camera model.